ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/chain/DefaultMutableBlockchain.java

```diff
@@ -171,6 +175,10 @@ private BlockAddedEvent appendBlockHelper(
     final BlockAddedEvent blockAddedEvent = updateCanonicalChainData(updater, block, td);
 
     updater.commit();
```

Comment:
Do we need to synchronize around the commit + chainHeader / totalDifficulty updates?

Reply:
They are called from the `appendBlock` method which is synchronized.